support/ruby/collins-notify/lib/collins_notify/adapter/slack.rb

```diff
@@ -0,0 +1,85 @@
+require 'json'
+
+module CollinsNotify
+
+  # currently this only supports incoming webhooks to pust to slack
+  # https://tumblr.slack.com/services/new/incoming-webhook
+
+  class SlackAdapter < Notifier
+    register_name :slack
+    supports_mimetype :text
+    require_config :webhook_url
+
+    def configure!
+      @webhook_url = config.adapters[:slack][:webhook_url]
+      logger.info "Configured Slack adapter"
+    end
+    
+    # Available in template binding:
+    # message_obj - Depends on call
+    # channel - channel sending to
+    def notify! message_obj = OpenStruct.new, to = nil
+      slack_hash = Hash.new
+      channel = get_channel config.adapters[:slack], to
+      slack_hash['channel'] = channel
+      optional_parameters = [:username, :icon_url, :icon_emoji]
+      optional_parameters.each do |op|
+        if config.adapters[:slack][op]
+          slack_hash[op] = config.adapters[:slack][op]
+        end
+      end
+
+      slack_hash['text'] = get_message_body(binding)
+
+      @logger.debug "slack parameters: #{slack_hash.inspect}"
+        
+      if config.test? then
+        @logger.info "Not sending message in test mode"
```

Comment:
Log what message you arent sending, instead of the slack parameters debug above?